syntax = "proto3";

package bank;

service BankService {
    rpc Query(CustomerRequest) returns (BalanceResponse);
    rpc Deposit(TransactionRequest) returns (EventResponse);
    rpc Withdraw(TransactionRequest) returns (EventResponse);
}

message CustomerRequest {
    int32 customer_id = 1;
}

message TransactionRequest {
    int32 customer_id = 1;
    int32 amount = 2;
    bool propagate = 3;
}

message BalanceResponse {
    int32 balance = 1;
}

message EventResponse {
    string status = 1;
}
